File: 6_kyu/CalculateStringRotation.swift
func shiftedDiff(_ first: String, _ second: String) -> Int? {
    guard first.count == second.count else {
        return nil
    }
    
    let combined = first + first
    for i in 0..<first.count {
        let index = combined.index(combined.startIndex, offsetBy: i)
        let substring = String(combined[index..<combined.index(index, offsetBy: first.count)])
        
        if substring == second {
            return i
        }
    }
    
    return nil
}

File: 6_kyu/CalculateStringRotation.cs
public class CalculateStringRotation
{
  public static int? ShiftedDiff(string first, string second)
  {
      if (first.Length != second.Length)
      {
          return null;
      }

      string combined = first + first;
      for (int i = 0; i < first.Length; i++)
      {
          string substring = combined.Substring(i, first.Length);
          if (substring == second)
          {
              return i;
          }
      }

      return null;
  }

}
